src/vcpkg/binarycaching.cpp

```diff
@@ -1716,6 +1768,43 @@ namespace vcpkg
         return results;
     }
 
+    void BinaryCache::push_thread_main()
+    {
+        decltype(actions_to_push) my_tasks;
+        while (true)
+        {
+            {
+                std::unique_lock<std::mutex> lock(actions_to_push_mutex);
+                actions_to_push_notifier.wait(lock, [this]() { return !actions_to_push.empty() || end_push_thread; });
+                if (actions_to_push.empty())
+                {
+                    if (end_push_thread) break;
+                    continue;
+                }
+
+                std::swap(my_tasks, actions_to_push);
+            }
+            // Now, consume all of `my_tasks` before taking the lock again.
+            for (auto& action_to_push : my_tasks)
+            {
+                if (end_push_thread)
+                {
+                    msg::println(msgUploadRemainingPackages, msg::count = remaining_packages_to_push);
+                }
+                for (auto&& provider : m_providers)
```

Comment:
We already emit messages for each push from the providers, right? I think this message is unneeded verbosity.

Reply:
What should we print if a provider has two sources and one fails and one is successful?

Reply:
We could also print
```
Stored binaries in {} destinations. 
```
because the files provider do not really upload binaries.

Reply:
It is now 
```
Wait until the remaining packages (9) are uploaded
Stored binaries in 2 destinations.  (1/9)
Stored binaries in 2 destinations.  (2/9)
Stored binaries in 2 destinations.  (3/9)
Stored binaries in 2 destinations.  (4/9)
Stored binaries in 2 destinations.  (5/9)
Stored binaries in 2 destinations.  (6/9)
Stored binaries in 2 destinations.  (7/9)
Stored binaries in 2 destinations.  (8/9)
Stored binaries in 2 destinations.  (9/9)
➜  vcpkg git:(qtbase-framework) ✗ 
```